### PR TITLE
Special holidays migration

### PR DESCRIPTION
**Pull request by arkid15r**
As #724 has been recently merged the next step is to migrate the rest of the countries to use the new approach.

Move logic from `_populate` to `special_holidays` for:
  - Botswana
  - Eswatini
  - Hongkong
  - Ireland
  - Japan
  - South Korea
  - Lesotho
  - Namibia
  - Poland
  - Singapore
  - Slovakia
  - South Africa
  - Zambia

### CODE REVIEW

File: holidays/countries/japan.py [hunk 1]
```diff
@@ -37,6 +37,9 @@ class Japan(HolidayBase):
     """
 
     country = "JP"
+    special_holidays = {
```
Comment (KJhellico) [inline holidays/countries/japan.py:40]: Japan has another special holidays (Regarding Emperor of Heisei and Emperor of Reiwa).
And Malaysia also (Malaysia General Election Holiday).

Comment (arkid15r) [reply holidays/countries/japan.py:40]: The suggestions have been addressed!

Thanks for reviewing this!

File: holidays/countries/singapore.py [hunk 2]
```diff
@@ -37,6 +37,12 @@
 
 class Singapore(HolidayBase):
     country = "SG"
+    special_holidays = {
+        # SG50 Public holiday
+        # Announced on 14 March 2015
+        # https://www.mom.gov.sg/newsroom/press-releases/2015/sg50-public-holiday-on-7-august-2015
+        2015: ((AUG, 7, "SG50 Public Holiday"),)
+    }
```
Comment (KJhellico) [inline holidays/countries/singapore.py:45]: There are Polling Day dates also

Comment (arkid15r) [reply holidays/countries/singapore.py:52]: It's great to have another pair of eyes! Thank you for reviewing!

Comment (dr-prodigy) [reply holidays/countries/singapore.py:52]: Thank you guys well done!

**Comment by dr-prodigy** [comment]
Merged 👍

**Pull request merged**